tracer/src/Datadog.Trace/ITracer.cs

```diff
@@ -0,0 +1,43 @@
+// <copyright file="ITracer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// The tracer is responsible for creating spans and flushing them to the Datadog agent
+    /// </summary>
+    public interface ITracer
+    {
+        /// <summary>
+        /// Gets the active scope
+        /// </summary>
+        IScope ActiveScope { get; }
+
+        /// <summary>
+        /// Gets the default service name for traces where a service name is not specified.
+        /// </summary>
+        string DefaultServiceName { get; }
+
+        /// <summary>
+        /// Gets this tracer's settings.
+        /// </summary>
+        ImmutableTracerSettings Settings { get; }
+
+        /// <summary>
+        /// This creates a new span with the given parameters and makes it active.
+        /// </summary>
+        /// <param name="operationName">The span's operation name</param>
+        /// <param name="parent">The span's parent</param>
+        /// <param name="serviceName">The span's service name</param>
+        /// <param name="startTime">An explicit start time for that span</param>
+        /// <param name="ignoreActiveScope">If set the span will not be a child of the currently active span</param>
+        /// <param name="finishOnClose">If set to false, closing the returned scope will not close the enclosed span </param>
+        /// <returns>A scope wrapping the newly created span</returns>
+        IScope StartActive(string operationName, ISpanContext parent = null, string serviceName = null, DateTimeOffset? startTime = null, bool ignoreActiveScope = false, bool finishOnClose = true);
```

Comment:
> why do we allow the setting service name here but not the resource name

My assumption is because `OperationName == ResourceName` in many cases? And during `Finish()` if we don't have a `ResourceName`, we set it to the `OperationName`.

Side point - I'd really like to improve the docs here. What **is** the operation name vs resource name. We need examples. I never remember and have to look what our integrations have done to see what's correct. `"The span's operation name"` is not useful 😉 

>The new span is a local root span and if you had a previous non-finished trace now you have multiple traces running in parallel? And hopefully you kept a reference to the previous active scope so you can finish it. 

Yes, exactly. I think this is the correct API for background tasks ([see this issue](https://github.com/DataDog/dd-trace-dotnet/issues/1285)) if you don't want them to be tied to the parent, e.g.

```csharp
using (Tracer.Instance.StartActive("outer-scope"))
{
    Task.Run(async () =>
    {
        using (Tracer.Instance.StartActive("also-outer-scope"))
        {
            Thread.Sleep(300);
        }
    });              
}
```

> I also have general concerns about optional parameters in public interfaces, but that's gonna be a harder sell.

I'm 100% with you, these should be overloads IMO 😉

Reply:
+1 on everything :)

Reply:
Created a ticket to finalize the public API of ITracer